docs/src/main/asciidoc/infinispan-client-guide.adoc

```diff
@@ -99,15 +103,93 @@ public class Book {
 }
 ----
 
-The default serialization is done using a library based on protobuf. We need to define the proto
-buf schema and a marshaller for each user type(s).
+Serialization of user types uses a library based on protobuf, called Protostream.
 
-NOTE: Annotation based proto stream marshalling is not yet supported in
-the Quarkus Infinispan client. This will be added soon, allowing you to only annotate your classes,
-skipping the following steps.
+=== Annotation based Serialization
 
-NOTE: This version has an issue currently requiring the Query DSL module as a dependency when running in
-native mode with proto buf based marshalling. This should be remedied in a future version.
+This can be done automatically by adding annotations to your user classes from the protostream library
+provided by Infinispan. In addition a single Initializer annoated interface is required
+which controls how the supporting classes are generated.
+
+Here is an example of how the preceding classes should be changed:
+
+.Author.java
+[source,java]
+----
+    @ProtoFactory
+    public Author(String name, String surname) {
+        this.name = Objects.requireNonNull(name);
+        this.surname = Objects.requireNonNull(surname);
+    }
+
+    @ProtoField(number = 1)
+    public String getName() {
+        return name;
+    }
+
+    @ProtoField(number = 2)
+    public String getSurname() {
+        return surname;
+    }
+----
+
+.Book.java
+[source,java]
+----
+    @ProtoFactory
+    public Book(String title, String description, int publicationYear, Set<Author> authors) {
+        this.title = Objects.requireNonNull(title);
+        this.description = Objects.requireNonNull(description);
+        this.publicationYear = publicationYear;
+        this.authors = Objects.requireNonNull(authors);
+    }
+
+    @ProtoField(number = 1)
+    public String getTitle() {
+        return title;
+    }
+
+    @ProtoField(number = 2)
+    public String getDescription() {
+        return description;
+    }
+
+    @ProtoField(number = 3, defaultValue = "1970")
```

Comment:
-1 is probably a better default to indicate the lack of a specified year. aka null :)

Reply:
Sure